### ISSUE TITLE
Whisper Feature Request: Subtitle Download Option

### ISSUE THREAD

**Issue by J1Man** [chenxwh/cog-sd-txt2imghd#3]
Hi Chenxi,

Thanks a lot for putting Openai Whisper to Replicate website.  It works great.  **I have a feature request.  Can you add the option to download the subtitle file generated by Whisper to your Replicate web implementation?**

The article that I linked below says that Whisper generates a VTT subtitle file that can be used for closed captioning.  The article says "The transcription is also saved to audio.wav.txt, along with a file audio.wav.vtt to be used for closed captioning".
https://www.assemblyai.com/blog/how-to-run-openais-whisper-speech-recognition-model/

I would appreciate if you can make it possible to download the VTT subtitle file generated by Whisper.  I use the translation option often, so if Whisper generates VTT files for both the original source language and the target translated language, I would appreciate if you can make VTT files for both languages downloadable. 

On a separate note, I am writing my request here because the issue tracker for your Whisper repository was closed.

Thanks for everything.

**Comment by J1Man** [comment]
Hi Chenxi,

There is another issue. Couple of hours ago I saw that you implemented subtitles, but before I had a chance to try it your Whisper implementation completely disappeared from Replicate website.  When I click the link below,
 https://replicate.com/cjwbw/whisper
It forwards to 
https://replicate.com/openai/whisper
and all your new changes, improvements and subtitle implementation is gone.  It seems like Openai signed up with Replicate and hijacked the model name "whisper".  They also hijacked all your initial web implementation.  Their page is identical to what you uploaded 4 days ago. 

If you submit your model with all your changes/patches to Replicate with a new name, it should resolve the issue.  Something like  "cjwbw/whisper-plus" or "cjwbw/whisper-withpatches" or "cjwbw/whisper-reloaded".  I am not good at coming up with names, but I just wanted to illustrate my point.   There are already multiple flavors of the same models implemented by different people on Replicate.  Just adding something to the name should be sufficient to show that it is your implementation with improvements.

**Comment by chenxwh** [comment]
Hi thanks for noticing the work!

We usually put the demo under our name first before we ask the actual owner of the repo to "claim" it, then we'd transfer to their account, so they can update the demo however they like including future versions etc., therefore the demo is under openai now.

Yes I did add the features for the subtitles, but it is then suggested that the demo should be "simple", and suggested that there are existing demo like this one https://replicate.com/m1guelpf/whisper-subtitles that output the subtitle maybe you can try? 

Additionally the .vtt file as output for downloading is not supported by Replicate yet (confirmed today after adding the subtitle files, but .srt and .txt work), so it is unfortunately not possible yet to easily download .vtt files even if I push the subtitle feature to a different demo page.

**Comment by J1Man** [comment]
Thanks for the clarifications.  I guess there is not much that can be done if they suggested to keep the demo simple without subtitles. 

SRT format subtitles is actually better for me.  My plan was to convert VTT subtitles to SRT anyway.  I did not know that SRT was supported by Whisper.   The news article that I read only mentioned VTT.

I did not know about m1guelpf's implementation.  Thanks for pointing it out.  I can use that since it has SRT subtitles feature.  I guess m1guelpf's implementation is my only option at this point. :)
